### DISCUSSION TITLE
How to do a wall-clock time aggregation

### DISCUSSION THREAD

**Discussion by pitermarx** [LGouellec/streamiz#236] [Q&A] [answered]
I've just realized that when doing an aggregation on a temporal window, what I get is an ever-updating IKTable of the values, and the windows are done via the stream time, and not wall-clock time.

My idea was to "group" all the events that happen on each minute and emit a stream of events that would have the aggregation of it, that only had a rate of 1 event per minute. (this might be complicated by multiple consumer groups, but let's imagine for now a single consumer on a single partition)

I'm seeing some options like supress and punctuate in the Java implementation but I'm not seeing a way out in the dotnet implementation.

2 questions
1 - Is there any way to do what I want in Streamiz?
2 - Does it even make sense to think in terms of wall-clock time when using KStreams? do people usually do it?

**Comment by LGouellec** [comment]
Hi @pitermarx , 

Unfortunately the wall-clock aggregation is not available yet.

1 - Is there any way to do what I want in Streamiz?
*Not possible for the moment*

2 - Does it even make sense to think in terms of wall-clock time when using KStreams? do people usually do it?
*Yes of course, the [feature](https://github.com/LGouellec/kafka-streams-dotnet/issues/199) is coming soon for 1.5.0*

**Comment by LGouellec** [comment, marked as answer]
Hi @pitermarx , 

Unfortunately the wall-clock aggregation is not available yet.

1 - Is there any way to do what I want in Streamiz?
*Not possible for the moment*

2 - Does it even make sense to think in terms of wall-clock time when using KStreams? do people usually do it?
*Yes of course, the [feature](https://github.com/LGouellec/kafka-streams-dotnet/issues/199) is coming soon for 1.5.0*

**Reply by pitermarx** [reply]
This is Gr8 news. Do you have a ballpark idea of when 1.5 will be published?

**Reply by LGouellec** [reply]
End Q2' 2023
